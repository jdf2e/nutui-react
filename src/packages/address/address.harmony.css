.nut-address-elevator {
  display: flex;
  margin-top: 20px;
}
.nut-address-exist {
  display: block;
  padding: 15px 20px 0;
  height: 279px;
  overflow-y: auto;
  box-sizing: border-box;
}
.nut-address-exist-item {
  display: flex;
  align-items: center;
  margin-bottom: 20px;
  font-size: 12px;
  line-height: 14px;
  color: #1a1a1a;
}
.nut-address-exist-item.active {
  font-weight: 500;
}
.nut-address-exist-item-info {
  margin-left: 9px;
}
.nut-address-footer {
  width: 100%;
  height: 54px;
  padding: 6px 0px 0;
  border-top: 1px solid rgba(0, 0, 0, 0.06);
}
.nut-address-footer-btn {
  width: 90%;
  height: 42px;
  line-height: 42px;
  margin: auto;
  text-align: center;
  background: linear-gradient(135deg, #f53d4d 0%, #fa2c19 100%);
  border-radius: 21px;
  font-size: 15px;
  color: #ffffff;
}

[dir=rtl] .nut-address-exist-item-info,
.nut-rtl .nut-address-exist-item-info {
  margin-left: 0;
  margin-right: 9px;
}