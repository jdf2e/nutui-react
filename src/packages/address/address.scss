@import '../popup/popup.scss';
@import '../elevator/elevator.scss';
@import '../cascader/cascader.scss';

.nut-address {
  display: block;

  &-header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-top: 24px;
    margin-bottom: 17px;
    padding: 0 20px;
    text-align: center;
    font-weight: bold;
    color: $title-color;
    line-height: 20px;

    &-title {
      display: block;
      color: $title-color;
      font-size: 18px;
    }
  }

  // 请选择
  display: block;

  &-tabs {
    position: relative;
    margin-top: 32px;
    padding: 0 20px;
    display: flex;
    font-size: 13px;
    color: $title-color;

    &-item {
      min-width: 2px;
      margin-right: 30px;
      display: block;

      &.active {
        clear: $title-color;
        font-weight: bold;
      }
    }

    &-line {
      position: absolute;
      bottom: -10px;
      left: 20px;
      display: inline-block;
      margin-top: 5px;
      width: 26px;
      height: 3px;
      background: $address-tab-line-background;
      transition: 0.2s all linear;
    }
  }

  .nut-address-custom {
    display: block;
    margin: 20px 20px 0;
    padding-top: 15px;
    height: 270px;
    overflow-y: auto;
    box-sizing: border-box;
    padding: 0;

    &-item {
      display: flex;
      align-items: center;
      margin-bottom: 20px;
      font-size: $font-size-1;
      color: $title-color;

      &.active {
        font-weight: bold;
        color: $title-color;
      }

      &-icon {
        margin-right: 6px;
      }
    }
  }

  .nut-address-elevator {
    display: flex;
    margin-top: 20px;
  }

  .nut-address-exist {
    display: block;
    margin-top: 15px;
    padding: 15px 20px 0;
    height: 279px;
    overflow-y: auto;
    box-sizing: border-box;

    &-item {
      display: flex;
      align-items: center;
      margin-bottom: 20px;
      font-size: $font-size-1;
      line-height: 14px;
      color: $title-color;

      &.active {
        font-weight: 500;
      }

      &-info {
        margin-left: 9px;
      }
    }
  }

  .nut-address-footer {
    width: 100%;
    height: 54px;
    padding: 6px 0px 0;
    border-top: 1px solid #f2f2f2;

    &-btn {
      width: 90%;
      height: 42px;
      line-height: 42px;
      margin: auto;
      text-align: center;
      background: $button-primary-background-color;
      border-radius: 21px;
      font-size: 15px;
      color: $primary-text-color;
    }
  }
}
