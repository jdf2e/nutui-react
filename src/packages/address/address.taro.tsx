import React, { FunctionComponent, useEffect, useState } from 'react'
import { Left } from '@nutui/icons-react-taro'
import Popup from '@/packages/popup/index.taro'
import { ExistRender } from './existRender.taro'
import { CustomRender } from './customRender.taro'
import { useConfig } from '@/packages/configprovider/configprovider.taro'
import { AddressList } from './type'
import {
  CascaderOption,
  CascaderOptionKey,
  CascaderProps,
  CascaderValue,
} from '@/packages/cascader/index.taro'
import { ComponentDefaults } from '@/utils/typings'

export interface AddressProps extends CascaderProps {
  visible: boolean
  value?: CascaderValue
  defaultValue?: CascaderValue
  type: string
  options: CascaderOption[]
  optionKey: CascaderOptionKey
  format: Record<string, string | number | null>
  custom: boolean | string
  existList: AddressList[]
  height: string | number
  defaultIcon: React.ReactNode
  selectIcon: React.ReactNode
  backIcon: React.ReactNode
  onSwitch?: (data: { type: string }) => void
  // 仅用于选择已有地址
  onExistSelect?: (data: AddressList) => void
}

const defaultProps = {
  ...ComponentDefaults,
  visible: false,
  defaultValue: [],
  type: 'custom',
  options: [],
  optionKey: { textKey: 'text', valueKey: 'value', childrenKey: 'children' },
  format: {},
  custom: false,
  existList: [],
  height: '200px',
  defaultIcon: null,
  selectIcon: null,
  closeIcon: null,
  backIcon: null,
} as unknown as AddressProps

export const Address: FunctionComponent<
  Partial<AddressProps> &
    Omit<
      React.HTMLAttributes<HTMLDivElement>,
      'onChange' | 'defaultValue' | 'onLoad' | 'title' | 'onClick'
    >
> = (props) => {
  const { locale } = useConfig()
  const {
    style,
    className,
    visible,
    defaultValue,
    children,
    type,
    options,
    optionKey,
    format,
    height,
    title,
    existList,
    custom,
    selectIcon,
    defaultIcon,
    closeIcon,
    backIcon,
    onChange,
    onExistSelect,
    onClose,
    onSwitch,

    ...rest
  } = {
    ...defaultProps,
    ...props,
  }

  const classPrefix = 'nut-address'
  const [currentType, setCurrentType] = useState<string>(type)
  const [showPopup, setShowPopup] = useState(visible)

  const handClose = () => {
    setShowPopup(false)
  }
  useEffect(() => {
    setShowPopup(visible)
  }, [visible])

  const renderLeftOnCustomSwitch = () => {
    return (
      <>
        {custom && (
          <div className={`${classPrefix}-left-icon`} onClick={onSwitchModule}>
            {React.isValidElement(backIcon) ? (
              backIcon
            ) : (
              <Left color="#cccccc" />
            )}
          </div>
        )}
      </>
    )
  }

  const selectedExistItem = (data: AddressList) => {
    onExistSelect && onExistSelect(data)
    handClose()
  }

  // 切换地址选择模式
  const onSwitchModule = () => {
    if (currentType === 'exist') {
      setCurrentType('custom')
    } else {
      setCurrentType('exist')
    }
    onSwitch && onSwitch({ type: currentType })
  }
  return (
    <>
      {currentType === 'custom' || currentType === 'custom2' ? (
        <CustomRender
          visible={showPopup}
          closeable
          title={title || locale.address.selectRegion}
          left={renderLeftOnCustomSwitch()}
          defaultValue={defaultValue}
          options={options}
          format={format}
          optionKey={optionKey}
          type={currentType}
          height={height}
          onClose={handClose}
          onChange={(val: CascaderValue, params?: any) => {
            onChange?.(val, params)
          }}
        />
      ) : (
        <Popup
          visible={showPopup}
          position="bottom"
          round
          closeable
          closeIcon={closeIcon}
          title={title || locale.address.selectRegion}
        >
          <div
            className={`${classPrefix} ${className || ''}`}
            style={{ ...style }}
          >
            {
              // 不需要 close，选中切换即关闭弹框。可手动关闭弹框，只关闭弹框不处理逻辑。
              <ExistRender
                type={currentType}
                existList={existList}
                selectIcon={selectIcon}
                defaultIcon={defaultIcon}
                custom={custom}
                onSelect={selectedExistItem}
                onSwitch={onSwitchModule}
              />
            }
          </div>
        </Popup>
      )}
    </>
  )
}

Address.defaultProps = defaultProps
Address.displayName = 'NutAddress'
