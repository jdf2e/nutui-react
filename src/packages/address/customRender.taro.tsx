import React, { FunctionComponent } from 'react'
import Cascader, {
  CascaderProps,
  CascaderOption,
  CascaderValue,
  CascaderOptionKey,
} from '@/packages/cascader/index.taro'
import { ComponentDefaults } from '@/utils/typings'

export interface AddressProps extends CascaderProps {
  visible: boolean // popup 显示状态
  type: string
  options: CascaderOption[]
  value?: CascaderValue
  defaultValue?: CascaderValue
  optionKey: CascaderOptionKey
  format: Record<string, string | number | null>
  height: string | number
}

const defaultProps = {
  ...ComponentDefaults,
  visible: false,
  type: 'custom',
  options: [],
  optionKey: { textKey: 'text', valueKey: 'value', childrenKey: 'children' },
  format: {},
  height: '200px',
} as unknown as AddressProps

export const CustomRender: FunctionComponent<
  Partial<AddressProps> &
    Omit<
      React.HTMLAttributes<HTMLDivElement>,
      'title' | 'defaultValue' | 'onChange'
    >
> = (props) => {
  const {
    children,
    visible,
    type,
    height,
    options,
    title,
    left,
    value,
    defaultValue,
    optionKey,
    format,
    onClose,
    onChange,
    onPathChange,
    ...rest
  } = {
    ...defaultProps,
    ...props,
  }

  return (
    <>
      {type === 'custom' && (
        <Cascader
          visible={visible}
          value={value}
          defaultValue={defaultValue}
          title={title}
          left={left}
          options={options}
          format={format}
          optionKey={optionKey}
          onClose={() => {
            onClose?.()
          }}
          onChange={(val) => {
            onChange?.(val)
          }}
          onPathChange={onPathChange}
          {...rest}
        />
      )}
    </>
  )
}
