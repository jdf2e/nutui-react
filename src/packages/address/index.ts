import { Address } from './address'
export default Address
