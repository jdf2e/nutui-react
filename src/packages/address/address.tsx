import React, { FunctionComponent, useEffect, useState } from 'react'
import { Left, Location2, Check, CircleClose } from '@nutui/icons-react'
import Popup from '@/packages/popup'
import { ExistRender } from './existRender'
import { CustomRender } from './customRender'
import { useConfig } from '@/packages/configprovider'
import {
  RegionData,
  NextList,
  Regions,
  ChangeData,
  ResultData,
  AddressList,
} from './type'

import { BasicComponent, ComponentDefaults } from '@/utils/typings'

export interface AddressProps extends BasicComponent {
  visible: boolean
  defaultValue: (string | number)[]
  type: string
  title: string
  province: RegionData[]
  city: RegionData[]
  country: RegionData[]
  town: RegionData[]
  custom: boolean | string
  existList: AddressList[]
  height: string | number
  defaultIcon: React.ReactNode
  selectIcon: React.ReactNode
  closeIcon: React.ReactNode
  backIcon: React.ReactNode
  onClose?: (data: ResultData) => void
  onSwitch?: (data: { type: string }) => void
  // 仅用于选择级联地址。
  onChange?: (data: ChangeData) => void
  onTabChecked?: (data: string) => void
  // 仅用于选择已有地址
  onSelect?: (data: AddressList) => void
}

const defaultProps = {
  ...ComponentDefaults,
  visible: false,
  defaultValue: [],
  type: 'custom',
  title: '',
  province: [],
  city: [],
  country: [],
  town: [],
  custom: false,
  existList: [],
  height: '200px',
  defaultIcon: null,
  selectIcon: null,
  closeIcon: null,
  backIcon: null,
} as AddressProps

export const Address: FunctionComponent<
  Partial<AddressProps> &
    Omit<
      React.HTMLAttributes<HTMLDivElement>,
      'onChange' | 'title' | 'onSelect' | 'defaultValue'
    >
> = (props) => {
  const { locale } = useConfig()
  const {
    visible,
    defaultValue,
    children,
    type,
    height,
    title,
    existList,
    province,
    city,
    country,
    town,
    custom,
    selectIcon,
    defaultIcon,
    closeIcon,
    backIcon,
    onChange,
    onSelect,
    onClose,
    onSwitch,
    onTabChecked,
    style,
    className,
    ...rest
  } = {
    ...defaultProps,
    ...props,
  }
  const classPrefix = 'nut-address'
  const [currentType, setCurrentType] = useState<string>(type)

  const [showPopup, setShowPopup] = useState(visible)
  const [selectedRegion, setSelectedRegion] = useState<Regions>({
    province: { name: '' },
    city: { name: '' },
    country: { name: '' },
    town: { name: '' },
  })

  const [selectedExistAddress, setSelectedExistAddress] = useState({}) // 当前选择的地址

  const handClose = () => {
    setShowPopup(false)
  }

  const nextAreaList = (item: NextList) => {
    const callbackParams = {
      next: item.next,
      value: item.value,
      custom: item.custom,
    }

    setSelectedRegion({
      ...(item.selectedRegion as typeof selectedRegion),
    })

    onChange && onChange(callbackParams)
  }

  const initAddress = () => {
    const tabName = ['province', 'city', 'country', 'town']
    for (let i = 0; i < tabName.length; i++) {
      setSelectedRegion({
        ...selectedRegion,
        [tabName[i]]: {},
      })
    }
  }
  const close = () => {
    const resCopy = {
      addressIdStr: '',
      addressStr: '',
      ...selectedRegion,
    }
    const res: ResultData = {
      data: {
        addressIdStr: '',
        addressStr: '',
        ...selectedRegion,
      },
      type: currentType,
    }
    if (currentType === 'custom' || currentType === 'custom2') {
      const { province, city, country, town } = resCopy
      resCopy.addressIdStr = [
        (province as RegionData).id || 0,
        (city as RegionData).id || 0,
        (country as RegionData).id || 0,
        (town as RegionData).id || 0,
      ].join('_')
      resCopy.addressStr = [
        (province as RegionData).name,
        (city as RegionData).name,
        (country as RegionData).name,
        (town as RegionData).name,
      ].join('')
      res.data = resCopy
    } else {
      res.data = selectedExistAddress as AddressList
    }

    initAddress()

    onClose && onClose(res)
  }

  // exist：已有地址列表
  const selectedExistItem = (data: AddressList) => {
    setSelectedExistAddress(data)
    onSelect && onSelect(data)
    handClose()
  }

  // 切换地址选择模式
  const onSwitchModule = () => {
    if (currentType === 'exist') {
      setCurrentType('custom')
    } else {
      setCurrentType('exist')
    }
    initAddress()
    onSwitch && onSwitch({ type: currentType })
  }

  const headerRender = () => {
    return (
      <div className={`${classPrefix}-header`}>
        <div className="arrow-back" onClick={onSwitchModule}>
          {currentType === 'custom' &&
            custom &&
            (React.isValidElement(backIcon) ? (
              backIcon
            ) : (
              <Left color="#cccccc" />
            ))}
        </div>

        <div className={`${classPrefix}-header-title`}>
          {title || locale.address.selectRegion}
        </div>

        <div onClick={() => handClose()}>
          {React.isValidElement(closeIcon) ? (
            closeIcon
          ) : (
            <CircleClose width={18} height={18} color="#ccc" />
          )}
        </div>
      </div>
    )
  }

  useEffect(() => {
    setShowPopup(visible)
  }, [visible])

  // useEffect(() => {
  //   if (!showPopup) {
  //     close()
  //   }
  // }, [showPopup])

  return (
    <Popup
      visible={showPopup}
      position="bottom"
      // onClose={() => {
      //   // 只需要处理关闭弹框，不需要处理任何逻辑。
      //   // 当再次打开时，以初始化数据为准。
      //   console.log('d-')
      //   // close()
      // }}
    >
      <div
        className={`${classPrefix} ${className || ''}`}
        style={{ ...style }}
        {...rest}
      >
        {headerRender()}
        {(currentType === 'custom' || currentType === 'custom2') && (
          <CustomRender
            defaultValue={defaultValue}
            type={currentType}
            province={province}
            city={city}
            country={country}
            town={town}
            height={height}
            onNextArea={(cal) => {
              nextAreaList && nextAreaList(cal)
            }}
            onTabClick={(type) => {
              onTabChecked && onTabChecked(type)
            }}
            onClose={handClose}
          />
        )}
        {currentType === 'exist' && (
          <ExistRender
            type={currentType}
            existList={existList}
            selectIcon={
              React.isValidElement(selectIcon) ? (
                selectIcon
              ) : (
                <Check color="#FA2C19" />
              )
            }
            defaultIcon={
              React.isValidElement(defaultIcon) ? defaultIcon : <Location2 />
            }
            custom={custom || (custom && locale.address.chooseAnotherAddress)}
            onSelect={selectedExistItem}
            onSwitchModule={onSwitchModule}
          />
        )}
      </div>
    </Popup>
  )
}

Address.defaultProps = defaultProps
Address.displayName = 'NutAddress'
