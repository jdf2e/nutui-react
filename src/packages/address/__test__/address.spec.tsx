import React from 'react'
import { render, waitFor, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Address } from '../address'

const addressData = {
  province: [
    { id: 1, name: '北京', title: 'B' },
    { id: 2, name: '广西', title: 'G' },
    { id: 3, name: '江西', title: 'J' },
    { id: 4, name: '四川', title: 'S' },
    { id: 5, name: '浙江', title: 'Z' },
  ],
  city: [
    { id: 7, name: '朝阳区', title: 'C' },
    { id: 8, name: '崇文区', title: 'C' },
    { id: 9, name: '昌平区', title: 'C' },
    { id: 6, name: '石景山区', title: 'S' },
    { id: 3, name: '八里庄街道', title: 'B' },
    { id: 10, name: '北苑', title: 'B' },
  ],
  country: [
    { id: 3, name: '八里庄街道', title: 'B' },
    { id: 9, name: '北苑', title: 'B' },
    { id: 4, name: '常营乡', title: 'C' },
  ],
  town: [],
}

const existAddress = [
  {
    id: 1,
    addressDetail: '',
    cityName: '次渠镇',
    countyName: '通州区',
    provinceName: '北京市',
    selectedAddress: true,
    townName: '',
    name: '探探鱼',
    phone: '182****1718',
  },
  {
    id: 2,
    addressDetail: '',
    cityName: '钓鱼岛全区',
    countyName: '',
    provinceName: '钓鱼岛',
    selectedAddress: false,
    townName: '',
    name: '探探鱼',
    phone: '182****1718',
  },
  {
    id: 3,
    addressDetail: '京东大厦',
    cityName: '大兴区',
    countyName: '科创十一街18号院',
    provinceName: '北京市',
    selectedAddress: false,
    townName: '',
    name: '探探鱼',
    phone: '182****1718',
  },
]

test('Show Address', async () => {
  const { container } = render(
    <Address
      modelValue
      province={addressData.province}
      city={addressData.city}
      country={addressData.country}
      town={addressData.town}
      customAddressTitle="请选择所在地区"
    />
  )
  const regionItem = container.querySelectorAll('.nut-address__region-item')
  expect(regionItem.length).toBe(5)
})

test('choose address item', async () => {
  const changeHandle = jest.fn()
  const { container } = render(
    <Address
      modelValue
      province={addressData.province}
      city={addressData.city}
      country={addressData.country}
      town={addressData.town}
      customAddressTitle="请选择所在地区"
      onChange={changeHandle}
    />
  )
  const regionItem = container.querySelectorAll('.nut-address__region-item')[0]
    .firstElementChild
  regionItem && fireEvent.click(regionItem)

  await waitFor(() => {
    const regionItemNext = container.querySelectorAll(
      '.nut-address__tab-item '
    )[0]
    expect(changeHandle.mock.calls[0][0].next).toEqual('city')
    expect(regionItemNext.textContent).toEqual('北京')
  })
})

test('default choose address', async () => {
  const changeHandle = jest.fn()
  const { container } = render(
    <Address
      modelValue
      modelSelect={[1, 7, 3]}
      province={addressData.province}
      city={addressData.city}
      country={addressData.country}
      town={addressData.town}
      customAddressTitle="请选择所在地区"
    />
  )
  const regionItem = container.querySelectorAll('.nut-address__region-tab')[0]
  const contentItem = container.querySelectorAll('.nut-address__region-item')[0]
  await waitFor(() => {
    expect(regionItem.textContent).toEqual('北京朝阳区请选择')
    expect(contentItem.querySelector('.nutui-iconfont')).toBeEmptyDOMElement()
  })
})

test('exist address', async () => {
  const { container } = render(
    <Address
      modelValue
      type="exist"
      existAddress={existAddress}
      isShowCustomAddress={false}
      customAddressTitle="请选择所在地区"
    />
  )
  const existItem = container.querySelectorAll('.nut-address__exist-item')

  await waitFor(() => {
    expect(existItem.length).toBe(3)
  })
})

test('exist address choose event', async () => {
  const selectHandle = jest.fn()
  const { container } = render(
    <Address
      modelValue
      type="exist"
      existAddress={existAddress}
      customAddressTitle="请选择所在地区"
      isShowCustomAddress={false}
      onSelected={selectHandle}
    />
  )
  const existSecondItem = container.querySelectorAll(
    '.nut-address__exist-item'
  )[1].firstElementChild
  existSecondItem && fireEvent.click(existSecondItem)

  await waitFor(() => {
    expect(selectHandle.mock.calls[0][1].id).toBe(2)
  })
})

test('exist address & list address', async () => {
  const changeHandle = jest.fn()
  const switchModule = jest.fn()
  const { container } = render(
    <Address
      modelValue
      type="exist"
      province={addressData.province}
      city={addressData.city}
      country={addressData.country}
      town={addressData.town}
      existAddress={existAddress}
      customAddressTitle="请选择所在地区"
      switchModule={switchModule}
      onChange={changeHandle}
    />
  )
  const chooseBtn = container.querySelectorAll(
    '.nut-address__choose-other-btn'
  )[0]
  chooseBtn && fireEvent.click(chooseBtn)

  await waitFor(() => {
    expect(switchModule).toBeCalled()
  })

  const regionItem = container.querySelectorAll('.nut-address__region-item')
  expect(regionItem.length).toBe(5)
})
