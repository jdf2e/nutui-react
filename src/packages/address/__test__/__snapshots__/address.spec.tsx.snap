// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address: exist defaultIcon & selectIcon 1`] = `"<div class="nut-overlay" style="--nutui-overlay-zIndex: 1000;"></div><div style="z-index: 1000;" class="round nut-popup nut-popup-bottom "><div class="nut-popup-title">请选择地址</div><div class="nut-popup__close-icon nut-popup__close-icon--top-right"><svg class="nut-icon nut-icon-Close " style="width: 12px; height: 12px;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024" aria-labelledby="Close" role="presentation"><path d="M865.83 926.17a42.67 42.67 0 1 0 60.36-60.34L572.35 512l353.84-353.83a42.67 42.67 0 0 0-60.36-60.34L512 451.67 158.19 97.83a42.67 42.67 0 0 0-60.36 60.34L451.67 512 97.83 865.83a42.67 42.67 0 1 0 60.34 60.34L512 572.33l353.83 353.84z" fill="currentColor" fill-opacity="0.9"></path></svg></div><div class="nut-address "><ul class="nut-address-exist"><li class="nut-address-exist-item "><div class="default">123</div><div class="nut-address-exist-item-info"><div>探探鱼</div><div>182****1718</div><div>北京市次渠镇通州区</div></div></li><li class="nut-address-exist-item "><div class="default">123</div><div class="nut-address-exist-item-info"><div>探探鱼</div><div>182****1718</div><div>钓鱼岛钓鱼岛全区</div></div></li><li class="nut-address-exist-item active"><div class="select">456</div><div class="nut-address-exist-item-info"><div>探探鱼</div><div>182****1718</div><div>北京市大兴区科创十一街18号院京东大厦</div></div></li></ul></div></div>"`;

exports[`Address: show custom 1`] = `"<div class="nut-overlay" style="--nutui-overlay-zIndex: 1000;"></div><div style="z-index: 1000;" class="round nut-popup nut-popup-bottom "><div class="nut-popup-left-icon"></div><div class="nut-popup-title">选择地址</div><div class="nut-popup__close-icon nut-popup__close-icon--top-right"><svg class="nut-icon nut-icon-Close " style="width: 12px; height: 12px;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024" aria-labelledby="Close" role="presentation"><path d="M865.83 926.17a42.67 42.67 0 1 0 60.36-60.34L572.35 512l353.84-353.83a42.67 42.67 0 0 0-60.36-60.34L512 451.67 158.19 97.83a42.67 42.67 0 0 0-60.36 60.34L451.67 512 97.83 865.83a42.67 42.67 0 1 0 60.34 60.34L512 572.33l353.83 353.84z" fill="currentColor" fill-opacity="0.9"></path></svg></div><div class="nut-cascader "><div class="nut-tabs nut-tabs--horizontal" value="c1"><div class="nut-tabs__titles nut-tabs__titles--line nut-tabs__titles--scrollable"><div class="nut-tabs__titles-item nut-tabs__titles-item--active"><span class="nut-tabs__titles-item__text">请选择</span><span class="nut-tabs__titles-item__line"></span></div></div><div class="nut-tabs__content__wrap"><div class="nut-tabs__content" style="transform: translate3d(--100%, 0, 0); transition-duration: 300ms;"><div class="nut-tabpane active"><div class="nut-cascader-pane"><div class="nut-cascader-item"><div class="nut-cascader-item__title">浙江</div></div><div class="disabled nut-cascader-item"><div class="nut-cascader-item__title">湖南</div></div><div class="nut-cascader-item"><div class="nut-cascader-item__title">福建</div></div></div></div></div></div></div></div></div>"`;

exports[`Address: show exist 1`] = `"<div class="nut-overlay" style="--nutui-overlay-zIndex: 1000;"></div><div style="z-index: 1000;" class="round nut-popup nut-popup-bottom "><div class="nut-popup-title">选择地址</div><div class="nut-popup__close-icon nut-popup__close-icon--top-right"><svg class="nut-icon nut-icon-Close " style="width: 12px; height: 12px;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024" aria-labelledby="Close" role="presentation"><path d="M865.83 926.17a42.67 42.67 0 1 0 60.36-60.34L572.35 512l353.84-353.83a42.67 42.67 0 0 0-60.36-60.34L512 451.67 158.19 97.83a42.67 42.67 0 0 0-60.36 60.34L451.67 512 97.83 865.83a42.67 42.67 0 1 0 60.34 60.34L512 572.33l353.83 353.84z" fill="currentColor" fill-opacity="0.9"></path></svg></div><div class="nut-address "><ul class="nut-address-exist"><li class="nut-address-exist-item active"><svg class="nut-icon nut-icon-Check " xmlns="http://www.w3.org/2000/svg" color="#FA2C19" viewBox="0 0 1024 1024" aria-labelledby="Check" role="presentation"><path d="M998.4 245.03c-219.43 153.6-398.63 332.8-552.23 552.23-40.23 58.51-128 54.86-164.57-3.66-69.49-106.06-149.94-186.51-256-256-51.2-32.91-18.29-113.37 40.23-98.74 117.03 21.94 208.46 69.49 292.57 146.28 157.26-190.17 358.4-340.11 588.8-435.2 62.17-25.6 106.06 58.51 51.2 95.09" fill="currentColor" fill-opacity="0.9"></path></svg><div class="nut-address-exist-item-info"><div>探探鱼</div><div>182****1718</div><div>北京市次渠镇通州区</div></div></li><li class="nut-address-exist-item "><svg class="nut-icon nut-icon-Location " xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024" aria-labelledby="Location" role="presentation"><path d="M682.67 470.04c0 94.4-76.42 170.94-170.67 170.94s-170.67-76.54-170.67-170.94S417.75 299.12 512 299.11s170.67 76.52 170.67 170.93zm-85.34 0A85.4 85.4 0 0 0 512 384.58c-47.15 0-85.33 38.27-85.33 85.46A85.4 85.4 0 0 0 512 555.5c47.15 0 85.33-38.25 85.33-85.46z" fill="currentColor" fill-opacity="0.9"></path>,<path d="M981.33 470.04c0 277.76-312.75 465.73-464.15 552.53a10.22 10.22 0 0 1-10.36 0C355.42 935.79 42.67 747.82 42.67 470.06 42.67 210.45 252.8 0 512 0s469.33 210.45 469.33 470.04zm-85.33 0c0-212.39-171.93-384.58-384-384.58S128 257.64 128 470.04c0 106.24 60.52 202.73 151.85 288.85 74.6 70.4 161.98 126.46 232.15 167.89 70.19-41.43 157.55-97.49 232.15-167.89C835.46 672.77 896 576.28 896 470.04z" fill="currentColor" fill-opacity="0.9"></path></svg><div class="nut-address-exist-item-info"><div>探探鱼</div><div>182****1718</div><div>钓鱼岛钓鱼岛全区</div></div></li><li class="nut-address-exist-item "><svg class="nut-icon nut-icon-Location " xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024" aria-labelledby="Location" role="presentation"><path d="M682.67 470.04c0 94.4-76.42 170.94-170.67 170.94s-170.67-76.54-170.67-170.94S417.75 299.12 512 299.11s170.67 76.52 170.67 170.93zm-85.34 0A85.4 85.4 0 0 0 512 384.58c-47.15 0-85.33 38.27-85.33 85.46A85.4 85.4 0 0 0 512 555.5c47.15 0 85.33-38.25 85.33-85.46z" fill="currentColor" fill-opacity="0.9"></path>,<path d="M981.33 470.04c0 277.76-312.75 465.73-464.15 552.53a10.22 10.22 0 0 1-10.36 0C355.42 935.79 42.67 747.82 42.67 470.06 42.67 210.45 252.8 0 512 0s469.33 210.45 469.33 470.04zm-85.33 0c0-212.39-171.93-384.58-384-384.58S128 257.64 128 470.04c0 106.24 60.52 202.73 151.85 288.85 74.6 70.4 161.98 126.46 232.15 167.89 70.19-41.43 157.55-97.49 232.15-167.89C835.46 672.77 896 576.28 896 470.04z" fill="currentColor" fill-opacity="0.9"></path></svg><div class="nut-address-exist-item-info"><div>探探鱼</div><div>182****1718</div><div>北京市大兴区科创十一街18号院京东大厦</div></div></li></ul></div></div>"`;
